src/vs/workbench/contrib/terminalContrib/accessibility/browser/textAreaSyncAddon.ts

```diff
@@ -100,7 +100,7 @@ export class TextAreaSyncAddon extends Disposable implements ITerminalAddon {
 			this._logService.debug(`TextAreaSyncAddon#updateCommandAndCursor: no line`);
 			return;
 		}
-		if (!!currentCommand.commandStartX) {
+		if (currentCommand?.commandStartX !== undefined) {
```

Comment:
Much better fix! You shouldn't need the `?` though as we have an early return on `!currentCommand` above:

```suggestion
		if (currentCommand.commandStartX !== undefined) {
```